### PR TITLE
Do the ABAP Language Version for the AFF objects

### PR DESCRIPTION
**Pull request by albertmink**
New behaviour:
* during serialization, the ABAP Language Version is taken from the repository (not from the object)
* during deserialization, the ABAP Language Version of the repository is set for all objects

Relates to #6154

### CODE REVIEW

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:247]:
```suggestion
        lo_object_handler = get_object_handler( ).
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:530]:
```suggestion
        lo_object_handler = get_object_handler( ).
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:602]:
```suggestion
        lv_pattern = ',\n\s*"abapLanguageVersion":\s"(cloudDevelopment|keyUser)"'.
        REPLACE FIRST OCCURRENCE OF REGEX lv_pattern IN lv_string WITH ''.
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:486]:
```suggestion
          lv_string                   TYPE string,
          lv_pattern                TYPE string,
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (mbtools) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:592]: You can use `zcl_abapgit_convert`, `string_to_xstring` `xstring_to_string` for this part

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:597]:
```suggestion
        lv_string = zcl_abapgit_convert=>xstring_to_string_utf8( iv_data = lv_json_as_xstring ).
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:609]:
```suggestion
          lv_json_as_xstring_wo_alv_x = zcl_abapgit_convert=>string_to_xstring_utf8( lv_string ).
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [reply src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:592]: alright, thanks

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:111]:
```suggestion
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:201]:
```suggestion
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:379]:
```suggestion
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:475]:
```suggestion
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (albertmink) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:495]:
```suggestion
          lv_file_as_xstring          TYPE xstring.
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (schneidermic0) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:596]: Should we move this to a dedicated (private) method?

**Comment by jfilak** [comment]
Boys, I am gonna do a little bit of trolling but those commit messages are little bit below the good software engineering standards, don't you think?

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (schneidermic0) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:589]: Maybe, we can also simplify the regular expression to be more stable for future ABAP language versions and other whitespaces

```suggestion
        lv_pattern = ',\s*"abapLanguageVersion"\s*:\s*"[^"]+"'.
```

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (Christopher-Hermann) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:636]: In my opinion the regex is too fragile. 
What happens if the abapLanguageVersion is the first field in the header, then the , will not match. 
What happens if the abapLanguageVersion is not contained in the header (the default is standard so it can be omitted). In this case we would match the next abapLanguageVersion field.

When you include the "header":{} into the regex, it should be more stable, what do you think?
`^"header"\s*:\s*\{.*?\s*("abapLanguageVersion"\s*:\s*"[^"]+").*?}`

Comment (mbtools) [inline src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:636]: This assumes a certain order but JSON generally does not guarantee that. I'm not sure why there should be other alav fields since we don't want to serialize them anyway

Comment (mbtools) [reply src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:636]: That doesn't handle the comma properly. The output must have a stable format and therefore also handle newlines (or there might be empty lines in the json output).

Something like this will be better:

```abap
data li_json type ref to zif_ajson.
li_json = zcl_ajson=>parse( json_string ).
li_json->delete( '/header/abapLanguageVersion` ).
json_string = li_json->stringyfy( iv_indent = 2 ).
```

Comment (schneidermic0) [reply src/objects/aff/zcl_abapgit_object_common_aff.clas.abap]: Using AJSON parser, now.

Comment (schneidermic0) [reply src/objects/aff/zcl_abapgit_object_common_aff.clas.abap]: Using AJSON parser instead of regex, now.

File: src/objects/aff/zcl_abapgit_object_common_aff.clas.abap
Comment (schneidermic0) [reply src/objects/aff/zcl_abapgit_object_common_aff.clas.abap:591]: I have added a bunch of unit tests

**Comment by schneidermic0** [comment]
Sorry, I have requested review too early. :/

Let me fix the abaplint I issues, first.

**Review by Christopher-Hermann — APPROVED** [review]
Looks good for me.

**Review by schneidermic0 — APPROVED** [review]
For me, it looks fine as well. I would like to wait for @mbtools feedback before I merge it.

**Comment by mbtools** [comment]
🇨🇦 holiday. Back Wednesday

**Comment by schneidermic0** [comment]
> 🇨🇦 holiday. Back Wednesday

No rush. Enjoy!

**Review by mbtools — APPROVED** [review]
Alright 👍

**Comment by larshp** [comment]
lets go, merging

**Pull request merged**
